### PR TITLE
Docs: Clarify audio keycodes

### LINKED ISSUE #4617: AU_TOG description in the docs is misleading
The docs say `Toggles Audio mode` when what it actually does it toggle the audio _state_ (on/off). Toggling the audio _mode_ sounds more like it would switch between different audio modes (of which there are a few now).

### LINKED ISSUE #4619: Audio docs point to non-existant file for midi information
https://docs.qmk.fm/#/feature_audio?id=midi-functionality points to `quantum/keymap_midi.c` which is not a thing that exists.

### PR DESCRIPTION
**Pull request by drashna**
Closes #4617 and closes #4619

### CODE REVIEW

File: docs/feature_audio.md [hunk 1]
```diff
@@ -61,9 +61,11 @@ It's advised that you wrap all audio features in `#ifdef AUDIO_ENABLE` / `#endif
...
 The available keycodes for audio are: 
 
-* `AU_ON` - Turn audio mode on
-* `AU_OFF` - Turn audio mode off
-* `AU_TOG` - Toggle audio mode
+* `AU_ON` - Turn audio state on
+* `AU_OFF` - Turn audio state off
+* `AU_TOG` - Toggle audio state
+
+!> These keycodes turn all of the audio functionaly on and off.  Turning it off means that audio feedback, audio clicky, music mode, etc are disabled, completely.
```
Comment (noroadsleft) [inline docs/feature_audio.md]:
```suggestion
!> These keycodes turn all of the audio functionality on and off.  Turning it off means that audio feedback, audio clicky, music mode, etc. are disabled, completely. 
```

File: docs/keycodes.md [hunk 2]
```diff
@@ -226,9 +226,9 @@ This is a reference only. Each group of keys links to the page documenting their
 
 |Key             |Aliases  |Description                       |
 |----------------|---------|----------------------------------|
-|`AU_ON`         |         |Audio mode on                     |
-|`AU_OFF`        |         |Audio mode off                    |
-|`AU_TOG`        |         |Toggles Audio mode                |
+|`AU_ON`         |         |Turns on Audio features           |
+|`AU_OFF`        |         |Turns off all Audio features      |
+|`AU_TOG`        |         |Toggles Audio state                |
```
Comment (noroadsleft) [inline docs/keycodes.md]:
```suggestion
|`AU_TOG`        |         |Toggles Audio state               |
```

File: docs/feature_audio.md (hunk 1 above)
Comment (drashna) [reply docs/feature_audio.md]: WTF my spellchecker didn't catch that :'(

Comment (jetpacktuxedo) [reply docs/feature_audio.md]: it is spelled right, it is just a different word :P

**Comment by jetpacktuxedo** [comment]
Honestly, you could probably just say `Toggles Audio` given that `MU_TOG` just says `Toggles Music Mode`. This also works, though imo.

File: docs/feature_audio.md (hunk 1 above)
Comment (noroadsleft) [reply docs/feature_audio.md]: Near as I know, `functionally` (two Ls) is the only accepted spelling. But yes, different word.

**Pull request merged**